server/src/main/java/org/apache/druid/indexing/overlord/IndexerMetadataStorageCoordinator.java

```diff
@@ -39,32 +39,15 @@
 import java.util.Set;
 
 /**
+ * Handles metadata transactions performed by the Overlord.
  */
 public interface IndexerMetadataStorageCoordinator
 {
   /**
-   * Retrieve all published segments which may include any data in the interval and are marked as used from the
-   * metadata store.
-   *
-   * The order of segments within the returned collection is unspecified, but each segment is guaranteed to appear in
-   * the collection only once.
-   *
-   * @param dataSource The data source to query
-   * @param interval   The interval for which all applicable and used segmented are requested.
-   * @param visibility Whether only visible or visible as well as overshadowed segments should be returned. The
-   *                   visibility is considered within the specified interval: that is, a segment which is visible
-   *                   outside of the specified interval, but overshadowed within the specified interval will not be
-   *                   returned if {@link Segments#ONLY_VISIBLE} is passed. See more precise description in the doc for
-   *                   {@link Segments}.
-   * @return The DataSegments which include data in the requested interval. These segments may contain data outside the
-   *         requested interval.
-   *
-   * @implNote This method doesn't return a {@link Set} because there may be an expectation that {@code Set.contains()}
-   * is O(1) operation, while it's not the case for the returned collection unless it copies all segments into a new
-   * {@link java.util.HashSet} or {@link com.google.common.collect.ImmutableSet} which may in turn be unnecessary in
-   * other use cases. So clients should perform such copy themselves if they need {@link Set} semantics.
+   * Retrieves all published segments that have partial or complete overlap with
+   * the given interval and are marked as used.
    */
-  default Collection<DataSegment> retrieveUsedSegmentsForInterval(
+  default Set<DataSegment> retrieveUsedSegmentsForInterval(
```

Comment:
We have essentially changed a long standing interface. 
This might break custom extensions. Should we call this out in dev notes ?

Reply:
I don't really feel so, because this interface is not exposed as an `@ExtensionPoint` or `@PublicApi` and extensions are not expected to implement this interface.

Reply:
Fair enough.